src/features/filter.ts

```diff
@@ -5,8 +5,20 @@ const initialState = {
   status: 'all',
 };
 
-export const filterSlice = createSlice({
+const filterSlice = createSlice({
   name: 'filter',
   initialState,
-  reducers: {},
+  reducers: {
+    setFilter: (value, action) => {
+      return { status: action.payload, query: value.query };
```

Comment:
In the `setFilter` reducer, the `status` is correctly set from `action.payload`, but the `query` should remain unchanged. The current implementation incorrectly assigns `value.query` to `query`, which is correct. Ensure that `query` is not altered in this reducer.